lib/matplotlib/tri/triinterpolate.py

```diff
@@ -408,8 +408,8 @@ def __init__(self, triangulation, z, kind='min_E', trifinder=None,
         self._tri_renum = tri_renum
         # Taking into account the node renumbering in self._z:
         node_mask = (node_renum == -1)
-        self._z[node_renum[~node_mask]] = self._z
         self._z = self._z[~node_mask]
+        self._z[node_renum[~node_mask]] = self._z
```

Comment:
The three lines can written more clearly as
~~~
valid_mask = (node_renum != -1)
self._z[node_renum[valid_mask]] = self._z[valid_mask]
~~~